src/corePlugins.js

```diff
@@ -1605,6 +1605,48 @@ export let corePlugins = {
       '.border-double': { 'border-style': 'double' },
       '.border-hidden': { 'border-style': 'hidden' },
       '.border-none': { 'border-style': 'none' },
+
+      '.border-x-solid': { 'border-left-style': 'solid', 'border-right-style': 'solid' },
+      '.border-x-dashed': { 'border-left-style': 'dashed', 'border-right-style': 'dashed' },
+      '.border-x-dotted': { 'border-left-style': 'dotted', 'border-right-style': 'dotted' },
+      '.border-x-double': { 'border-left-style': 'double', 'border-right-style': 'double' },
+      '.border-x-hidden': { 'border-left-style': 'hidden', 'border-right-style': 'hidden' },
+      '.border-x-none': { 'border-left-style': 'none', 'border-right-style': 'none' },
+
+      '.border-y-solid': { 'border-top-style': 'solid', 'border-bottom-style': 'solid' },
+      '.border-y-dashed': { 'border-top-style': 'dashed', 'border-bottom-style': 'dashed' },
+      '.border-y-dotted': { 'border-top-style': 'dotted', 'border-bottom-style': 'dotted' },
+      '.border-y-double': { 'border-top-style': 'double', 'border-bottom-style': 'double' },
+      '.border-y-hidden': { 'border-top-style': 'hidden', 'border-bottom-style': 'hidden' },
+      '.border-y-none': { 'border-top-style': 'none', 'border-bottom-style': 'none' },
+
+      '.border-b-solid': { 'border-bottom-style': 'solid' },
+      '.border-b-dashed': { 'border-bottom-style': 'dashed' },
+      '.border-b-dotted': { 'border-bottom-style': 'dotted' },
+      '.border-b-double': { 'border-bottom-style': 'double' },
+      '.border-b-hidden': { 'border-bottom-style': 'hidden' },
+      '.border-b-none': { 'border-bottom-style': 'none' },
+
+      '.border-l-solid': { 'border-left-style': 'solid' },
+      '.border-l-dashed': { 'border-left-style': 'dashed' },
+      '.border-l-dotted': { 'border-left-style': 'dotted' },
+      '.border-l-double': { 'border-left-style': 'double' },
+      '.border-l-hidden': { 'border-left-style': 'hidden' },
+      '.border-l-none': { 'border-left-style': 'none' },
+
+      '.border-r-solid': { 'border-right-style': 'solid' },
+      '.border-r-dashed': { 'border-right-style': 'dashed' },
+      '.border-r-dotted': { 'border-right-style': 'dotted' },
+      '.border-r-double': { 'border-right-style': 'double' },
+      '.border-r-hidden': { 'border-right-style': 'hidden' },
+      '.border-r-none': { 'border-right-style': 'none' },
```

Comment:
You may also want to add `border-s-*` and `border-e-*` classes for logical properties. This would line up with the border width utilities: https://tailwindcss.com/docs/border-width#using-logical-properties

Reply:
@crswll added